lib/models/addon.js

```diff
@@ -501,7 +531,7 @@ Addon.prototype.compileAddon = function(tree) {
   }
 
   var addonJs = this.processedAddonJsFiles(tree);
-  var templatesTree = this.compileTemplates(this._treeFor('addon-templates'));
+  var templatesTree = this.compileTemplates(tree);
```

Comment:
We need to look for templates across the addons tree to find templates in pods.  The previous scope was too narrow (addon/templates)